packages/flame_rive/example/lib/main.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:math';
+
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame/input.dart';
+import 'package:flame/palette.dart';
+import 'package:flame_rive/flame_rive.dart';
+import 'package:flutter/material.dart';
+import 'package:rive/rive.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  late final game = RiveExampleGame();
+
+  @override
+  Widget build(BuildContext context) {
+    return GameWidget(
+      game: game,
+    );
+  }
+}
+
+class RiveExampleGame extends FlameGame with HasTappableComponents {
+  @override
+  Color backgroundColor() {
+    return const Color(0xFFFFFFFF);
+  }
+
+  @override
+  Future<void>? onLoad() async {
+    await super.onLoad();
+    final skillsArtboard =
+        await loadArtboard(RiveFile.asset('assets/skills.riv'));
+    add(SkillsAnimationComponent(skillsArtboard));
+  }
+}
+
+class SkillsAnimationComponent extends RiveComponent with Tappable {
+  SkillsAnimationComponent(Artboard artboard)
+      : super(
+          artboard: artboard,
+          size: Vector2.all(550),
+        );
+
+  SMIInput<double>? _levelInput;
+
+  @override
+  Future<void>? onLoad() {
+    final controller = StateMachineController.fromArtboard(
+      artboard,
+      "Designer's Test",
+    );
+    if (controller != null) {
+      artboard.addController(controller);
+      _levelInput = controller.findInput<double>('Level');
+      _levelInput?.value = 0;
+    }
+    return super.onLoad();
+  }
+
+  @override
+  bool onTapDown(TapDownInfo info) {
+    final levelInput = _levelInput;
+    if (levelInput == null) {
+      return false;
+    }
+    levelInput.value = (levelInput.value + 1) % 3;
+    return true;
+  }
+}
+
+class Square extends PositionComponent with HasGameRef<RiveExampleGame> {
```

Comment:
This could extend from `RectangleComponent` to avoid code duplication.

Reply:
Which code is being duplciated?

Reply:
The render method, but `I think you can skip this for now since I'm refactoring the `ShapeComponent`s atm anyways, so I can fix it later.

Reply:
Most of that is ported from the rives lib render objects, I would hate to have different behaviors.

Reply:
What? The only thing it does is `canvas.drawRect(size.toRect(), paint);`, which will be the same as what `SquareComponent` does.